deploy/manifests/base/custom-types.yaml

```diff
@@ -42,6 +42,16 @@ spec:
           type: boolean
         config:
           type: object
+        configFrom:
+          type: object
+          properties:
+            secretKeyRef:
+              type: object
+              properties:
+                name:
+                  type: string
+                key:
+                  type: string
```

Comment:
How does this work for nested objects, where items are required if that object is present, but are not otherwise? fda7b3875ee1b730eaefb9be80b8f18e1bf955c7 is an attempt but not based on much concrete info--it's accepted, but I don't think there's any K8S-level validation below `properties`, as it's just a generic object.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#jsonschemaprops-v1beta1-apiextensions-k8s-io only has `required` at the top level, and it's not clear what the syntax is for nested items (presumably `configFrom.secretKeyRef.namespace`, but not sure). Regardless, `configFrom` isn't required, but things under it are if you use it, and it's not clear how to represent that: properties is just an object. 932c9b21d58e465f046d8f00595211a4e89e2cf4 uses the format in https://swagger.io/docs/specification/describing-parameters/, which seems like it's probably correct.

Reply:
What have you done looks correct to me from an OAS specification standpoint. This should work recursively with CRDs, if not, it is a bug.